### ISSUE TITLE
Kibana missing from repo

### ISSUE THREAD

**Issue by nealef** [openshift/origin-aggregated-logging#1469]
While building aggregated logging images for 3.11 the kibana build fails as it cannot find the package in the repo described by kibana.repo: https://artifacts.elastic.co/packages/5.x/yum. The only two packages in that repo are elasticsearch and logstash.

**Comment by nealef** [comment]
Build output. clefos:latest is my version of openshift/base-centos7 as I am building for s390x (I should check these are as similar as possible - do you know where the build for this is?) 
```
[openshift/origin-logging-kibana5] --> FROM clefos:latest as 0
[openshift/origin-logging-kibana5] --> MAINTAINER OpenShift Development <dev@lists.openshift.redhat.com>
[openshift/origin-logging-kibana5] --> EXPOSE 5601
[openshift/origin-logging-kibana5] --> ENV ELASTICSEARCH_URL=https://logging-es:9200     HOME=/opt/app-root/src     KIBANA_BIN=/usr/share/kibana/bin/kibana     KIBANA_CONF_DIR=/etc/kibana     KIBANA_HOME=/usr/share/kibana     KIBANA_VER=5.6.10     RELEASE_STREAM=origin
[openshift/origin-logging-kibana5] --> LABEL io.k8s.description="Kibana container for querying Elasticsearch for aggregated logs"       io.k8s.display-name="Kibana"       io.openshift.expose-services="5601:http"       io.openshift.tags="logging,elk,kibana"
[openshift/origin-logging-kibana5] --> ADD kibana.repo /etc/yum.repos.d/kibana.repo
[openshift/origin-logging-kibana5] --> RUN INSTALLED_PKGS="kibana-${KIBANA_VER}" &&     yum install -y --setopt=tsflags=nodocs  ${INSTALLED_PKGS} &&     yum clean all
[openshift/origin-logging-kibana5] Loaded plugins: fastestmirror, ovl
[openshift/origin-logging-kibana5] Determining fastest mirrors
[openshift/origin-logging-kibana5] No package kibana-5.6.10 available.
[openshift/origin-logging-kibana5] Error: Nothing to do
[openshift/origin-logging-kibana5] running 'INSTALLED_PKGS="kibana-${KIBANA_VER}" &&     yum install -y --setopt=tsflags=nodocs  ${INSTALLED_PKGS} &&     yum clean all' failed with exit code 1
[ERROR] PID 3928: hack/build-images.sh:56: `return "${result}"` exited with status 1.
```
Note, when I get it install kibana from another repo it proceeds to the point where install looks for the origin-kibana zip file. However, I can't determine where this file is obtained - I know it comes from https://github.com/openshift/origin-kibana/releases but I don't see where the file is being retrieved.

**Comment by richm** [comment]
ah - I'm not familiar with 390
Here is the kibana.repo file we are using: https://github.com/openshift/origin-aggregated-logging/blob/release-3.11/kibana/kibana.repo - is this not current/correct?  you said `Note, when I get it install kibana from another repo it proceeds` - what other repo are you using?  The origin-kibana.zip file is here - https://github.com/openshift/origin-aggregated-logging/blob/release-3.11/kibana/lib/origin-kibana-v5.6.13-1.zip
Are you trying to build with `Dockerfile` or `Dockerfile.centos7`?  You will have to use the latter

**Comment by nealef** [comment]
Ah, I realize what's happened. When I built 3.10 I hand-built the kibana RPM as the one in the repo is not noarch like elasticsearch. So I will need to do it again with the level specified by the Dockerfile. 

How does that zip file get made available to the build process? I can't see it being added in the Dockerfile.

**Comment by richm** [comment]
@jcantrill how is the kibana  https://github.com/openshift/origin-aggregated-logging/blob/release-3.11/kibana/lib/origin-kibana-v5.6.13-1.zip built?

**Comment by richm** [comment]
> How does that zip file get made available to the build process? I can't see it being added in the Dockerfile.

Here: https://github.com/openshift/origin-aggregated-logging/blob/master/kibana/Dockerfile.centos7#L29
```
ADD lib/* ${HOME}/
```

**Comment by nealef** [comment]
D'oh. I had -1 and not -4!

Thanks. I will give it another try with my new RPM and see how things go this time. Thanks for your help. I'll update with the success (or otherwise) of this build.

**Comment by richm** [comment]
thanks!

**Comment by nealef** [comment]
Could you point me to the Dockerfile for openshift/base-centos7? I want to make one identical with s390x binaries.

**Comment by richm** [comment]
https://hub.docker.com/r/openshift/base-centos7/
which says it comes from
https://github.com/openshift/sti-base/
but I don't see where it builds an image tagged with `openshift/base-centos7`
@stevekuznetsov do you know how https://hub.docker.com/r/openshift/base-centos7/ is built and tagged?

**Comment by stevekuznetsov** [comment]
No, I do not. @bparees or @smarterclayton might know?

**Comment by bparees** [comment]
i don't think it is built anymore.  The SCL team took over and renamed it.  Hence why that image was last pushed 2 years ago.  You want centos/s2i-base-centos7 now.

**Comment by richm** [comment]
oh fun - we'll need to change all of our upstream logging images then . . .

**Comment by richm** [comment]
no, just kibana

**Comment by nealef** [comment]
Getting closer. I'l try and understand what the complaint is:
```
[openshift/origin-logging-kibana5] Plugin installation was unsuccessful due to error "Optimizations failure.
[openshift/origin-logging-kibana5]     ERROR in /usr/share/kibana/src/ui/public/autoload/styles.js
[openshift/origin-logging-kibana5]     Module not found: Error: Cannot resolve 'file' or 'directory' ../../../../ui_framework/dist/ui_framework.css in /usr/share/kibana/src/ui/public/autoload
[openshift/origin-logging-kibana5]      @ /usr/share/kibana/src/ui/public/autoload/styles.js 4:0-57
[openshift/origin-logging-kibana5]     
[openshift/origin-logging-kibana5]     ERROR in /usr/share/kibana/src/core_plugins/kibana/public/visualize/listing/visualize_listing.js
[openshift/origin-logging-kibana5]     Module not found: Error: Cannot resolve module 'ui_framework/services' in /usr/share/kibana/src/core_plugins/kibana/public/visualize/listing
[openshift/origin-logging-kibana5]      @ /usr/share/kibana/src/core_plugins/kibana/public/visualize/listing/visualize_listing.js 14:16-48
```
The only difference I can see is I am using nodejs 8 rather than 6.

**Comment by nealef** [comment]
> i don't think it is built anymore. The SCL team took over and renamed it. Hence why that image was last pushed 2 years ago. You want centos/s2i-base-centos7 now.

Thanks. I have that built.

**Comment by richm** [comment]
> The only difference I can see is I am using nodejs 8 rather than 6.

That could be the problem

**Comment by nealef** [comment]
I fixed the RPM build (it wasn't installing the ui_framework and plugin directories). Got a clean build this time. Thanks again for your help.

At the moment my build process for origin, registry, webconsole, and logging is done via a script. I would eventually like to duplicate the build infrastructure so that I can more quickly enable s390x support (the diffs are very very small). Any advice on where to start learning about the process used?

**Comment by tbielawa** [comment]
@jcantrill For tracking/visibility, can you send a note to our mailinglist/JIRA with the details of what you're asking about? If it's something ART can help with, then that will help us divide and conquer!

**Comment by nealef** [comment]
Will do. Is this the correct mailing list: dev@lists.openshift.redhat.com<mailto:dev@lists.openshift.redhat.com>?



@jcantrill<https://github.com/jcantrill> For tracking/visibility, can you send a note to our mailinglist/JIRA with the details of what you're asking about? If it's something ART can help with, then that will help us divide and conquer!

**Comment by richm** [comment]
> Will do. Is this the correct mailing list: dev@lists.openshift.redhat.com<mailto:dev@lists.openshift.redhat.com>?

yes

**Comment by tbielawa** [comment]
Work Requests aos-art-requests@redhat.com
